File: src/e-community/android/compose/app/src/main/java/at/fhooe/ecommunity/data/remote/signalr/dto/MeterDataRTDto.kt
package at.fhooe.ecommunity.data.remote.signalr.dto

import java.util.*

/**
 * meter data
 * @param smartMeterId id of smart meter
 * @param memberId id of member
 * @param activePowerPlus current power P+
 * @param activePowerMinus current power P-
 * @param reactivePowerPlus current power R+
 * @param reactivePowerMinus current power R-
 */
data class MeterDataRTDto<T> (var smartMeterId: UUID,
                              var memberId: UUID,
                              var activePowerPlus: T,
                              var activePowerMinus: T,
                              var reactivePowerPlus: T,
                              var reactivePowerMinus: T,)

File: src/e-community/backend/cloud/e-community-cloud/e-community-cloud-lib/Endpoints/Dtos/MeterDataRTDto.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace e_community_cloud_lib.Endpoints.Dtos
{
    /// <summary>
    /// real time meter data
    /// </summary>
    public class MeterDataRTDto
    {
        /// <summary>
        /// id of smart meter
        /// </summary>
        public Guid SmartMeterId { get; set; }

        /// <summary>
        /// id of member
        /// </summary>
        public Guid MemberId { get; set; }

        /// <summary>
        /// current power P+
        /// </summary>
        public int ActivePowerPlus { get; set; }

        /// <summary>
        /// current power P-
        /// </summary>
        public int ActivePowerMinus { get; set; }

        /// <summary>
        /// current power R+
        /// </summary>
        public int ReactivePowerPlus { get; set; }

        /// <summary>
        /// current power R-
        /// </summary>
        public int ReactivePowerMinus { get; set; }
    }
}
